core/zoe/utils.js

```diff
@@ -0,0 +1,32 @@
+// used to reduce boolean arrays
+const bothTrue = (prev, curr) => prev && curr;
+const anyTrue = (prev, curr) => prev || curr;
+
+// https://stackoverflow.com/questions/17428587/transposing-a-2d-array-in-javascript/41772644#41772644
+const transpose = matrix =>
```

Comment:
Do you actually need both representations, or could you just keep it in the transposed form?

Reply:
We only need the transposed version to do the conservation of rights check, because that's summing by issuer column, whereas most everything else is concerned with a player perspective (the rows). Since we don't actually store both, I think this is ok. Did you think this would give bad performance?